README.md

```diff
@@ -18,6 +18,9 @@ Then alter your homeserver configuration, adding to your `modules` configuration
 modules:
   - module: synapse_domain_rule_checker.DomainRuleChecker
     config:
+      # Use regex for all domain comparisons
+      use_regex: false
+
```

Comment:
I'm not sure I understand why this needs to be a config option that people opt in to. Does this break some kind of backwards compatibility? 
 - E.g. are there characters like that might appear in a domain that have a special meaning in a regex? AFAIK none of `[]()|*+{}^$\` can appear in a domain, but I haven't checked that thoroughly.
 - Or is today's behaviour something like a glob match? That would have a different meaning if interpreted as a regex.

If it does need to be a config option, we should spell out what happens when the option is false (i.e. exact matches only).

Reply:
The biggest issue is the `.` character which is contained in most domain names :) 

regex `foo.matrix.com` matches both `foo.matrix.com` and for example `fooxmatrix.com` - which would make all previously configured rule checkers inherently vulnerable.

Another thing (Although not that bad) is that former dict match that goes through hash table is way faster than going over all regexes and matching them one by one, so I assumed it would be better to make it opt-in just for when it's needed.

I will improve the comment though

Reply:
D'oh. Thanks, I completely forgot about `.`.